x/farming/types/plan.go

```diff
@@ -364,9 +364,13 @@ func IsPlanActiveAt(plan PlanI, t time.Time) bool {
 	return !plan.GetStartTime().After(t) && plan.GetEndTime().After(t)
 }
 
-// PrivatePlanFarmingPoolAddress returns a unique farming pool address
-// for a newly created plan.
-func PrivatePlanFarmingPoolAddress(name string, planId uint64) sdk.AccAddress {
-	poolAddrName := strings.Join([]string{PrivatePlanFarmingPoolAddrPrefix, fmt.Sprint(planId), name}, PoolAddrSplitter)
-	return address.Module(ModuleName, []byte(poolAddrName))
+// PrivatePlanFarmingPoolAcc returns a unique farming pool address for a newly created plan.
+func PrivatePlanFarmingPoolAcc(name string, planId uint64) sdk.AccAddress {
+	poolAccName := strings.Join([]string{PrivatePlanFarmingPoolAccPrefix, fmt.Sprint(planId), name}, AccNameSplitter)
+	return DeriveAddress(ReserveAccAddressType, ModuleName, poolAccName)
+}
+
+// StakingReserveAcc returns module account for the staking reserve pool account by staking coin denom and type.
+func StakingReserveAcc(stakingCoinDenom string) sdk.AccAddress {
+	return DeriveAddress(ReserveAccAddressType, ModuleName, StakingReserveAccPrefix+AccNameSplitter+stakingCoinDenom)
```

Comment:
Can we introduce new constant named `StakingReserveAccAddressType`(or `StakingReserveAddressType`) rather than using `ReserveAccAddressType`?

Reply:
Since `ReserveAddressType` is also used in `PrivatePlanFarmingPoolAcc`, the general `ReserveAddressType` would be better.

Reply:
Oh, sorry. I thought it was `ReserveRewardAddressType`, not `ReserveAddressType`.